File: 프로그래머스/unrated/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<string> solution(string myString) {
    vector<string> answer;
    string temp = "";
    
    for (int i = 0; i < myString.size(); i++)
    {
        if (myString[i] == 'x' && temp != "")
        {
            answer.push_back(temp);
            temp = "";
        }
        else if (myString[i] != 'x')
            temp += myString[i];
    }
    if (temp.size() >= 1)
        answer.push_back(temp);
    
    sort(answer.begin(), answer.end());
    
    return answer;
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.java
import java.util.Arrays;

class Solution {
    public String[] solution(String myString) {
        String[] tmp = myString.split("x");
        Arrays.sort(tmp);
        return Arrays.stream(tmp)
                .filter(str -> str.length() >= 1)
                .toArray(String[]::new);
    }
}